daemon/main.c

```diff
@@ -512,10 +514,20 @@ post_phase_init(
         }
 
         //  Now it's time to catch signals.
+        int signals[] = { SIGTERM, SIGCHLD, SIGHUP };
+        sigset_t mask;
+        unsigned sig;
+        struct sigaction action = { 0 };
+        action.sa_handler = sigcatch;
+        sigemptyset(&mask);
+
+        for (sig = 0; sig < sizeof(signals) / sizeof(signals[0]); sig++) {
+            sigaddset(&mask, sig);
+
+            sigaction(sig, &action, NULL);
+        }
 
-        sigset(SIGTERM, sigcatch);
-        sigset(SIGCHLD, sigcatch);
-        sigset(SIGHUP, sigcatch);
+        sigprocmask(SIG_UNBLOCK, &mask, NULL);
```

Comment:
This is changing the behaviour. Why do you want this?

Reply:
There is a warning that `sigset` is deprecated/obsolete.

According to the manpage 'sig is removed from the process's signal mask', so this tries to keep the behaviour the same.